### PR TITLE
Add Constructor Support for Solang Solidity Contracts on Soroban

### PR DESCRIPTION
**Pull request by tareknaser**
This PR is the first step towards implementing #1672. It adds constructor support for Solidity contracts in Solang. The next step, not covered in this PR, is handling cases where the user defines a `__constructor` function with custom logic. More details can be found in the issue discussion.

### Changes Made
- Renamed the `init` function (used to call `storage_initializers`) to `__constructor` so that it is recognized and invoked by the Soroban host environment.
- Updated tests and CI to work with Soroban version 22.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       uses: actions/checkout@v4
       with:
         submodules: recursive
-    - uses: dtolnay/rust-toolchain@1.75.0
+    - uses: dtolnay/rust-toolchain@1.81.0
```
Comment (salaheldinsoliman) [inline .github/workflows/release.yml:18]: Why is that change related to the PR?

**Review by salaheldinsoliman — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (tareknaser) [reply .github/workflows/release.yml:18]: `soroban-sdk` 22.0.0-rc.3.2 requires rustc 1.79.0 or newer, and `soroban-cli` (which we use in the CI) now requires rustc 1.81.0 or newer.

**Review by seanyoung — APPROVED** [review]
LGTM
